packages/next/server/web/sandbox/context.ts

```diff
@@ -0,0 +1,240 @@
+        mapExports: { Response: 'Response' },
+      },
+      {
+        path: require.resolve('../spec-compliant/request'),
+        mapExports: { Request: 'Request' },
+      },
+    ],
+  })
+
+  const moduleCache = {
+    context: context,
+    paths: new Map<string, string>(),
+    require: requireCache,
+    warnedEvals: new Set<string>(),
+  }
+
+  context.__next_eval__ = function __next_eval__(fn: Function) {
+    const key = fn.toString()
+    if (!moduleCache?.warnedEvals.has(key)) {
+      const warning = new Error(
+        `Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Middleware`
+      )
+      warning.name = 'DynamicCodeEvaluationWarning'
+      Error.captureStackTrace(warning, __next_eval__)
+      moduleCache?.warnedEvals.add(key)
+      options.onWarning(warning)
+    }
+    return fn()
+  }
+
+  context.fetch = (input: RequestInfo, init: RequestInit = {}) => {
+    init.headers = new Headers(init.headers ?? {})
+    const prevs = init.headers.get(`x-middleware-subrequest`)?.split(':') || []
+    const value = prevs.concat(options.module).join(':')
+    init.headers.set('x-middleware-subrequest', value)
+    init.headers.set(`user-agent`, `Next.js Middleware`)
+
+    if (typeof input === 'object' && 'url' in input) {
+      return fetch(input.url, {
+        ...init,
+        headers: {
+          ...Object.fromEntries(input.headers),
+          ...Object.fromEntries(init.headers),
+        },
+      })
+    }
+
+    return fetch(String(input), init)
+  }
+
+  return moduleCache
+}
+
+/**
+ * Create a base context with all required globals for the runtime that
+ * won't depend on any externally provided dependency.
+ */
+function createContext() {
+  const context: { [key: string]: unknown } = {
+    _ENTRIES: {},
+    atob: polyfills.atob,
+    Blob,
+    btoa: polyfills.btoa,
+    clearInterval,
+    clearTimeout,
+    console: {
+      assert: console.assert.bind(console),
+      error: console.error.bind(console),
+      info: console.info.bind(console),
+      log: console.log.bind(console),
+      time: console.time.bind(console),
+      timeEnd: console.timeEnd.bind(console),
+      timeLog: console.timeLog.bind(console),
+      warn: console.warn.bind(console),
+    },
+    CryptoKey: polyfills.CryptoKey,
+    Crypto: polyfills.Crypto,
+    crypto: new polyfills.Crypto(),
+    File,
+    FormData,
+    process: { env: { ...process.env } },
+    ReadableStream: polyfills.ReadableStream,
+    setInterval,
+    setTimeout,
+    TextDecoder,
+    TextEncoder,
+    TransformStream,
+    URL,
+    URLSearchParams,
+
+    // Indexed collections
+    Array,
+    Int8Array,
+    Uint8Array,
+    Uint8ClampedArray,
+    Int16Array,
+    Uint16Array,
+    Int32Array,
+    Uint32Array,
+    Float32Array,
+    Float64Array,
+    BigInt64Array,
+    BigUint64Array,
+
+    // Keyed collections
+    Map,
+    Set,
+    WeakMap,
+    WeakSet,
+
+    // Structured data
+    ArrayBuffer,
+    SharedArrayBuffer,
```

Comment:
Why are these listed here? shouldn't the context have it's own variants of them?

Reply:
It was already there before but I think it's simply to be explicit. If at some point we want to tweak one of those implementations we can simply substitute it there.